### PR TITLE
Add Consul Template

### PR DESCRIPTION
**Pull request by galal-hussein**
This template will create a secure consul cluster, it will create 3 containers and it requires the user to add the following variables:

- ca certificate.
- first consul certificate and key.
- second consul certificate and key.
- third consul certificate and key.
- gossip encryption key.

The certificates will secure the rpc communication between consul containers by enabling the native TLS encryption features in Consul, the containers will bind to the rancher's assigned IP.

### CODE REVIEW

**Comment by cloudnautique** [comment]
Is there an SVG of the Consul logo?

A readme might be nice, with some links to container source...and basic instructions.

**Comment by galal-hussein** [comment]
Added the README file and the image squashed, unfortunately couldn't find SVG image for the logo.

**Comment by cloudnautique** [comment]
I had no luck finding an SVG as well.

File: templates/consul/0/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ 
+### Usage:
+ 
+ Select Consul from catalog.
+
+ Enter the certificates and keys for consul nodes, ca certificates, and the encryption key.
+
+ Click deploy.
+ 
+ The consul nodes will be binded to the Rancher's assigned IPs.
```
Comment (cloudnautique) [inline templates/consul/0/README.md]: Might read better as:
The Consul nodes will be bound to the Rancher managed network IPs.

**Comment by cloudnautique** [comment]
OK, lets squash and get this merged.

**Comment by galal-hussein** [comment]
Done.

**Comment by galal-hussein** [comment]
Thanks!

**Pull request merged**